File: SWEA/D3/1208. ［S／W 문제해결 기본］ 1일차 － Flatten/［S／W 문제해결 기본］ 1일차 － Flatten.py
for t in range(1,11):
    dump = int(input())
    boxes = [int(i) for i in input().split()]
    for i in range(dump):
        max_v = max(boxes)
        min_v = min(boxes)
        if max_v - min_v <= 1:
            print(f'#{t} {max_v - min_v}')
            break
        max_idx = boxes.index(max_v)
        min_idx = boxes.index(min_v)
        boxes[max_idx] -= 1
        boxes[min_idx] += 1
    else:
        print(f'#{t} {max(boxes) - min(boxes)}')

File: SWEA/D3/1208. ［S／W 문제해결 기본］ 1일차 － Flatten/［S／W 문제해결 기본］ 1일차 － Flatten.java
import java.util.Scanner;
import java.io.FileInputStream;

class Solution {
    public static void main(String args[]) throws Exception {
        Scanner sc = new Scanner(System.in);

        for (int test_case = 1; test_case <= 10; test_case++) {

            int dump = sc.nextInt();
            int[] boxes = new int[100];
            for (int i = 0; i < 100; i++) {
                boxes[i] = sc.nextInt();
            }

            for (int d = 0; d < dump; d++) {
                int max_v = boxes[0];
                int min_v = boxes[0];
                for (int i : boxes) {
                    if (i > max_v) max_v = i;
                    if (i < min_v) min_v = i;
                }

                int gap = max_v - min_v;
                if (gap <= 1) {
                    break;
                } else {
                    int max_idx = 0;
                    int min_idx = 0;
                    for (int i = 0; i < boxes.length; i++) {
                        if (boxes[i] == max_v) {
                            max_idx = i;
                        } else if (boxes[i] == min_v) {
                            min_idx = i;
                        }
                    }
                    boxes[max_idx] -= 1;
                    boxes[min_idx] += 1;
                }
            }
            int max_v = boxes[0];
            int min_v = boxes[0];
            for (int i : boxes) {
                if (i > max_v) max_v = i;
                if (i < min_v) min_v = i;
            }

            int gap = max_v - min_v;
            System.out.println("#" + test_case + " " + gap);
        }
    }
}